docs/getting-started/assets/signature_checking/check-validator-up.sh

```diff
@@ -38,29 +42,50 @@ while [[ $# > 0 ]]; do
             PUBKEY="$1"
             shift 1
         ;;
+        --bechpubkey)
+            shift 1
+            BECH_PUBKEY="$1"
+            shift 1
+        ;;
         *)
             echoerr "Unknown argument: $1"
-            exit 1
         ;;
     esac
 done
 set +u
+if [[ ! -z "${BECH_PUBKEY}" ]]; then
+    check_chain_maind
+    if [[ "${RET_VALUE}" != 0 ]]; then
+        echoerr "curl is not installed. Please install it first for key conversion"
+    fi
+    PUBKEY=$(chain-maind debug pubkey ${BECH_PUBKEY} 2>&1 | grep "tendermint/PubKeyEd25519" | cut -d : -f2- | jq -r .value || echoerr "Decode Pubkey Failed ❌")
+fi
 if [[ -z "${PUBKEY}" ]]; then
 	echoerr "Missing --pubkey {base64 encoded Tendermint public key}"
-	exit 1
 fi
 set -u
-ADDRESS=$(curl --max-time 10 -sSL "${TENDERMINT_URL}/validators?per_page=300" | jq -r --arg PUBKEY "${PUBKEY}" '.result.validators[] | select(.pub_key.value == $PUBKEY) | .address')
+NUM=1
+while true; do
+    ERR=$(curl -sSL "${TENDERMINT_URL}/validators?per_page=100&page=${NUM}" | jq -r .error)
+    if [[ $ERR == "null" ]]; then 
```

Comment:
null is returned if it's an empty page?

Reply:
null is returned if it is still within pagination.
it is not null when tendermint rpc return
`bash
{
  "jsonrpc": "2.0",
  "id": -1,
  "error": {
    "code": -32603,
    "message": "Internal error",
    "data": "page should be within [1, 2] range, given 3"
  }
}
`